### PR TITLE
Add passkey creation to the onboarding flow [sc-16248]

### PR DESCRIPTION
**Pull request by vitaliipustovyi**

### CODE REVIEW

File: apps/nfid-frontend/src/features/authentication/services.ts [hunk 1]
```diff
@@ -81,3 +84,19 @@ export const checkIf2FAEnabled = async (context: AuthenticationContext) => {
...
+
+  try {
+    const hasPasskeys = await passkeyConnector.hasPasskeys()
+    return { showPasskeys: !hasPasskeys }
+  } catch (_) {
+    return { showPasskeys: true }
+  }
+}
+
+export const shouldShowSNSBanner = async () => {
```
Comment (olekid) [inline apps/nfid-frontend/src/features/authentication/services.ts:99]: do we need this if it's possible to check in guard?

File: apps/nfid-frontend/src/features/authentication/auth-selection/passkey-flow/services.ts [hunk 2]
```diff
@@ -121,12 +121,27 @@ export class PasskeyConnector {
...
+    if (!data?.length) throw new Error("No devices found")
+    return data
+  }
+
+  async hasPasskeys(): Promise<boolean> {
+    try {
+      const devices = await this.getDevices()
+      return devices[0].some((d) => d.device === "Keychain")
+    } catch (_) {
+      throw new Error("No passkeys found")
```
Comment (olsemeno) [inline apps/nfid-frontend/src/features/authentication/auth-selection/passkey-flow/services.ts:136]: some unexpected error. Wrong text

**Review by olsemeno — APPROVED** [review]

**Review by olekid — APPROVED** [review]

File: apps/nfid-frontend/src/features/authentication/root/root-machine.typegen.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@
 export interface Typegen0 {
   "@@xstate/typegen": true
   internalEvents: {
+    "": { type: "" }
```
Comment (olekid) [inline apps/nfid-frontend/src/features/authentication/root/root-machine.typegen.ts:6]: ?

File: apps/nfid-frontend/src/features/authentication/root/root-machine.typegen.ts [hunk 4]
```diff
@@ -76,7 +66,7 @@ export interface Typegen0 {
     isExistingAccount: "done.invoke.AuthWithGoogleMachine"
     isReturn: "done.invoke.AuthWithEmailMachine"
     showPasskeys: "done.invoke.shouldShowPasskeys"
-    showSNSBanner: "done.invoke.shouldShowSNSBanner"
+    showSNSBanner: ""
```
Comment (olekid) [inline apps/nfid-frontend/src/features/authentication/root/root-machine.typegen.ts:69]: ?

File: apps/nfid-frontend/src/features/authentication/root/root-machine.typegen.ts (hunk 3 above)
Comment (vitaliipustovyi) [reply apps/nfid-frontend/src/features/authentication/root/root-machine.typegen.ts:6]: AI's resonse:
The empty event ("": { type: "" }) can happen with an always transition (which is an implicit transition) that doesn't require an event type.
It's a placeholder for internal transitions and typically doesn't affect your state machine behavior.
You can safely ignore it unless you want to dive into customizing the typegen output.

Comment (vitaliipustovyi) [reply apps/nfid-frontend/src/features/authentication/root/root-machine.typegen.ts:6]: what do you think?) @olekid

**Pull request merged**